client/state/current-user/gravatar-status/actions.js

```diff
@@ -20,20 +20,13 @@ import {
 	withAnalytics,
 } from 'state/analytics/actions';
 
-export function uploadGravatar( file, bearerToken, email ) {
+export function uploadGravatar( file, email ) {
 	return dispatch => {
 		dispatch( withAnalytics(
 			recordTracksEvent( 'calypso_edit_gravatar_upload_start' ),
 			{ type: GRAVATAR_UPLOAD_REQUEST }
 		) );
-
-		const data = new FormData();
-		data.append( 'filedata', file );
-		data.append( 'account', email );
-		return request
-			.post( 'https://api.gravatar.com/v1/upload-image' )
-			.send( data )
-			.set( 'Authorization', 'Bearer ' + bearerToken )
+		return wpcom.undocumented().uploadGravatar( email, file )
```

Comment:
I think we are trying to move away from using `wpcom` and use the [data layer](https://github.com/Automattic/wp-calypso/tree/master/client/state/data-layer) instead. Maybe this is a good time to switch `uploadGravatar` over? cc @gwwar and @dmsnell for advice on this.